botorch/utils/transforms.py

```diff
@@ -124,19 +124,52 @@ def normalize_indices(indices: Optional[List[int]], d: int) -> Optional[List[int
     return normalized_indices
 
 
+def _verify_output_shape(cls: Any, X: Tensor, output: Tensor) -> bool:
+    r"""
+    Performs the output shape checks for `t_batch_mode_transform`. Output shape checks
+    help in catching the errors due to AcquisitionFunction arguments with erroneous
+    return shapes before these errors propagate further down the line.
+
+    This method checks that the `output` shape matches either the t-batch shape of X
+    or the `batch_shape` of `cls.model`.
+
+    Args:
+        cls: The AcquisitionFunction object being evaluated.
+        X: The `... x q x d`-dim input tensor with an explicit t-batch.
+        output: The return value of `cls.method(X, ...)`.
+
+    Returns:
+        True if `output` has the correct shape, False otherwise.
+    """
+    try:
+        return (
+            output.shape == X.shape[:-2]
+            or (output.shape == torch.Size() and X.shape[:-2] == torch.Size([1]))
+            or output.shape == cls.model.batch_shape
+        )
+    except (AttributeError, NotImplementedError):
+        # cls does not have model or cls.model does not define `batch_shape`
+        return False
```

Comment:
So I don't know if returning `False` here is the right thing to do. Basically if we get to this point, we have made some assumptions about the acquisition function (that it operates on a model that implements a `batch_shape` property) that are not true. I think maybe being somewhat lenient here and returning `True` but raising a `RuntimeWarning` pointing this out instead may be the better approach (as in less annoying to work around if doing custom stuff)?

Reply:
I am thinking in what scenario you'd get to here. Either you'd use one of the canned acquisition functions (that uses `t_batch_mode_transform`) with a custom model that doesn't define `batch_shape`, or you wrote a custom class, used `t_batch_model_transform` in it without passing `assert_output_shape = False`. In the second scenario, it is easy to fix by passing `assert_output_shape = False`, but the first one would require editing the source or redefining the acquisition function, which is annoying. Yeah, ignoring a `RuntimeWarning` sounds less annoying.